### PR TITLE
feat: update multiple files in e2e node with framework helpers

### PR DESCRIPTION
**Pull request by draveness**
/kind cleanup
/priority backlog
/sig testing
/sig node 

**What this PR does / why we need it**:

Update multiple files in e2e node with framework helpers

**Which issue(s) this PR fixes**:

ref: #79686

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by draveness** [comment]
cc/ @oomichi

File: test/e2e_node/eviction_test.go [hunk 1]
```diff
@@ -637,7 +637,8 @@ func verifyEvictionEvents(f *framework.Framework, testSpecs []podEvictSpec, expe
 			}.AsSelector().String()
 			podEvictEvents, err := f.ClientSet.CoreV1().Events(f.Namespace.Name).List(metav1.ListOptions{FieldSelector: selector})
 			gomega.Expect(err).To(gomega.BeNil(), "Unexpected error getting events during eviction test: %v", err)
-			gomega.Expect(len(podEvictEvents.Items)).To(gomega.Equal(1), "Expected to find 1 eviction event for pod %s, got %d", pod.Name, len(podEvictEvents.Items))
+			gomega.Expect(err).To(gomega.BeNil(), "Unexpected error getting events during eviction test: %v", err)
```
Comment (k-toyoda-pi) [inline test/e2e_node/eviction_test.go]: unnecessary line?

File: test/e2e_node/gpu_device_plugin_test.go [hunk 2]
```diff
@@ -116,20 +116,21 @@ var _ = framework.KubeDescribe("NVIDIA GPU Device Plugin [Feature:GPUDevicePlugi
 			ginkgo.By("Checking that scheduled pods can continue to run even after we delete device plugin.")
 			ensurePodContainerRestart(f, p1.Name, p1.Name)
 			devIdRestart1 = parseLog(f, p1.Name, p1.Name, deviceIDRE)
+			framework.ExpectEqual(devIdRestart1, devId1)
 			gomega.Expect(devIdRestart1).To(gomega.Equal(devId1))
```
Comment (k-toyoda-pi) [inline test/e2e_node/gpu_device_plugin_test.go]: I think that `gomega.Expect...` at line:120 is unnecesarry.

File: test/e2e_node/eviction_test.go (hunk 1 above)
Comment (draveness) [reply test/e2e_node/eviction_test.go]: done

File: test/e2e_node/gpu_device_plugin_test.go (hunk 2 above)
Comment (draveness) [reply test/e2e_node/gpu_device_plugin_test.go]: done

**Comment by draveness** [comment]
/retest

Regards,
Draven

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

Thanks for this clean up :)

**Comment by mattjmcnaughton** [comment]
/test pull-kubernetes-integration

I think this is flake, but if its a true failure, please ping me after fixing and I'll mark "lgtm" after the fix is merged :)

**Comment by draveness** [comment]
> /test pull-kubernetes-integration
> 
> I think this is flake, but if its a true failure, please ping me after fixing and I'll mark "lgtm" after the fix is merged :)

It's a flake, and thanks for your review

**Comment by oomichi** [comment]
Thanks

/approve

**Pull request merged**